### PR TITLE
refactor validate input

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

File: web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
-import Intl from '../../../i18n/i18n';
 import { parseForm } from '../../../utils/parseForm';
 import { useState, ChangeEvent, FormEvent } from 'react';
 import {
   formatStandard,
-  reverseFormatStandard,
-  checkRecipientFormat,
   Unit,
+  validate,
+  validateAmount,
```
Comment (mazmassa) [inline web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:7]: What's the difference between them!?

Comment (Pierregvx) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:7]: validate checks both the amount and the recipient

Comment (mazmassa) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:7]: Ok! Great! In this case why we have validateAmount then!? 🤔

Comment (Pierregvx) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:7]: For the selection of the fees, there is no field for recipient there

Comment (mazmassa) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:7]: Humm... I see! Ok! 

In future we must to review anyway how we improve that.

**Review by mazmassa — APPROVED** [review]

File: web-frontend/src/utils/MassaFormating.tsx [hunk 2]
```diff
@@ -67,3 +71,74 @@ export function maskAddress(str: string, length = 4, mask = '. . .'): string {
...
+}
+
+/**
+ * Validates a recipient address.
+ * @param amount - The amount to validate.
+ * @param balance - The balance to validate.
+ * @returns An object containing the error message if the address is invalid, `null` otherwise.
+ * @see validate
+ */
+
```
Comment (Thykof) [inline web-frontend/src/utils/MassaFormating.tsx:103]:
```suggestion
```

File: web-frontend/src/utils/MassaFormating.tsx [hunk 3]
```diff
@@ -67,3 +71,74 @@ export function maskAddress(str: string, length = 4, mask = '. . .'): string {
...
+/**
+ * Validates a recipient address.
+ * @param recipient - The recipient address to validate.
+ * @param amount - The amount to validate.
+ * @param balance - The balance to validate.
+ * @returns An object containing the error message if the address is invalid, `null` otherwise.
+ */
+
+export function validate(
```
Comment (Thykof) [inline web-frontend/src/utils/MassaFormating.tsx:83]: the tsdoc, the name of the function are not in accordance to what the function really do

File: web-frontend/src/utils/MassaFormating.tsx
Comment (Thykof) [inline web-frontend/src/utils/MassaFormating.tsx:1]: for validation functions, we have a validation folder in scr

**Review by Thykof — CHANGES_REQUESTED** [review]

File: web-frontend/src/pages/TransferCoins/ReceiveCoins/GenerateLink.tsx [hunk 4]
```diff
@@ -13,6 +13,12 @@ import {
   PopupModalHeader,
   Selector,
 } from '@massalabs/react-ui-kit';
+import { validateInputs } from '../../../validation/sendInputs';
+
+interface IError {
```
Comment (Thykof) [inline web-frontend/src/pages/TransferCoins/ReceiveCoins/GenerateLink.tsx:18]: rename this interface to something like SendInputErrors.

declare this interface in ../../../validation/sendInputs and import it here, modify the return type of validateInputs to this interface.

File: web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx [hunk 5]
```diff
@@ -23,56 +18,33 @@ function Send(props: SendProps) {
...
+    const { amount, recipient } = formObject;
+    const errors = validateInputs(
+      amount,
+      recipient,
+      'recipient',
+      unformattedBalance,
+    );
+    setError(errors);
+    if (errors !== null) return;
+    setValid(!valid);
```
Comment (Thykof) [inline web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:45]: why `valid = !valid`?

why not `valid = true`

File: web-frontend/src/validation/sendInputs.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  }
+  if (Number.isNaN(amountNum)) {
+    return {
+      amount: Intl.t('errors.send.invalid-amount', { type: amountType }),
+    };
+  }
+
+  console.log(balance);
+  if (!balance) return null;
+  console.log(balance);
```
Comment (Thykof) [inline web-frontend/src/validation/sendInputs.tsx:56]:
```suggestion
  if (!balance) return null;
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx (hunk 5 above)
Comment (Pierregvx) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:45]: I don't really get why yet but it didn't worked when we tried ( cc @pivilartisant  )

Comment (pivilartisant) [reply web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx:49]: either works

**Review by Thykof — APPROVED** [review]

**Pull request merged**